SpatialGDK/Source/SpatialGDK/Private/Tests/LoadBalanceEnforcerTest.cpp

```diff
@@ -0,0 +1,358 @@
+
+	AddLBEntityToView(View, EntityIdOne, ThisRequirementSet, OtherVirtualWorker);
+
+	SpatialGDK::FSubView SubView(SpatialConstants::LB_TAG_COMPONENT_ID, SpatialGDK::FSubView::NoFilter, &View, Dispatcher,
+								 SpatialGDK::FSubView::NoDispatcherCallbacks);
+
+	TArray<SpatialGDK::EntityComponentUpdate> Updates;
+
+	SpatialGDK::SpatialLoadBalanceEnforcer LoadBalanceEnforcer = SpatialGDK::SpatialLoadBalanceEnforcer(
+		ThisWorker, SubView, VirtualWorkerTranslator.Get(), [&Updates](SpatialGDK::EntityComponentUpdate Update) {
+			Updates.Add(MoveTemp(Update));
+		});
+
+	LoadBalanceEnforcer.Advance();
+
+	LoadBalanceEnforcer.ShortCircuitMaybeRefreshAcl(EntityIdOne);
+
+	bool bSuccess = true;
+	if (Updates.Num() == 1)
+	{
+		bSuccess &= Updates[0].EntityId == EntityIdOne;
+		bSuccess &= Updates[0].Update.GetComponentId() == SpatialConstants::ENTITY_ACL_COMPONENT_ID;
+		WriteAclMap AclMap = GetWriteAclMapFromUpdate(Updates[0]);
+		bSuccess &= AclMapDelegatesComponents(AclMap, OtherRequirementSet, NonAclLBComponents);
+	}
+	else
+	{
+		bSuccess = false;
+	}
+
+	TestTrue("LoadBalanceEnforcer returned expected ACL assignment results", bSuccess);
+
+	return true;
+}
+
+LOADBALANCEENFORCER_TEST(GIVEN_authority_intent_change_op_WHEN_we_inform_load_balance_enforcer_THEN_send_acl_update)
+{
+	SpatialGDK::FDispatcher Dispatcher;
+	SpatialGDK::EntityView View;
+	SpatialGDK::ViewDelta Delta;
+	const TUniquePtr<SpatialVirtualWorkerTranslator> VirtualWorkerTranslator = CreateVirtualWorkerTranslator();
+
+	AddLBEntityToView(View, EntityIdOne, OtherRequirementSet, ThisVirtualWorker);
+
+	SpatialGDK::FSubView SubView(SpatialConstants::LB_TAG_COMPONENT_ID, SpatialGDK::FSubView::NoFilter, &View, Dispatcher,
+								 SpatialGDK::FSubView::NoDispatcherCallbacks);
+
+	TArray<SpatialGDK::EntityComponentUpdate> Updates;
+
+	SpatialGDK::SpatialLoadBalanceEnforcer LoadBalanceEnforcer = SpatialGDK::SpatialLoadBalanceEnforcer(
+		ThisWorker, SubView, VirtualWorkerTranslator.Get(), [&Updates](SpatialGDK::EntityComponentUpdate Update) {
+			Updates.Add(MoveTemp(Update));
+		});
+
+	PopulateViewDeltaWithComponentUpdated(
+		Delta, View, EntityIdOne,
+		MakeComponentUpdateFromUpdate(SpatialGDK::AuthorityIntent::CreateAuthorityIntentUpdate(ThisVirtualWorker)));
+	SubView.Advance(Delta);
+	LoadBalanceEnforcer.Advance();
+
+	bool bSuccess = true;
+	if (Updates.Num() == 1)
+	{
+		bSuccess &= Updates[0].EntityId == EntityIdOne;
+		bSuccess &= Updates[0].Update.GetComponentId() == SpatialConstants::ENTITY_ACL_COMPONENT_ID;
+		WriteAclMap AclMap = GetWriteAclMapFromUpdate(Updates[0]);
+		bSuccess &= AclMapDelegatesComponents(AclMap, ThisRequirementSet, NonAclLBComponents);
+	}
+	else
+	{
+		bSuccess = false;
+	}
+
+	TestTrue("LoadBalanceEnforcer returned expected ACL assignment results", bSuccess);
+
+	return true;
+}
+
+LOADBALANCEENFORCER_TEST(GIVEN_component_presence_change_op_WHEN_we_advance_load_balance_enforcer_THEN_acl_update_contains_new_components)
```

Comment:
there's only one test for each of component presence and net owning client changes. This is because we're just testing the ACL is built correctly, we tested short circuiting above.